ciongke/cmd/test-pr/main.go

```
@@ -0,0 +1,392 @@
+		log.Printf("Failed to create source archive: %s", err)
+		return
+	}
+	if err = gcs.UploadSource(*sourceBucket, *head, sourcePath); err != nil {
+		log.Printf("Failed to upload source: %s", err)
+		return
+	}
+
+	b, err := ioutil.ReadFile(filepath.Join(clonePath, ".test.yml"))
+	if err != nil {
+		log.Printf("Cannot read .test.yml: %s", err)
+		return
+	}
+	jobs := []jobDescription{}
+	if err = yaml.Unmarshal(b, &jobs); err != nil {
+		log.Printf("Cannot unmarshal .test.yml: %s", err)
+		return
+	}
+
+	config, err := restclient.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Error getting rest client: %s", err)
+	}
+	cl, err := client.New(config)
+	if err != nil {
+		log.Fatalf("Error getting client: %s", err)
+	}
+	batchClient, err := client.NewBatch(config)
+	if err != nil {
+		log.Fatalf("Error getting batch client: %s", err)
+	}
+	testClients := testClient{
+		GitHubClient:    githubClient,
+		KubeClient:      cl,
+		KubeBatchClient: batchClient,
+	}
+
+	var wg sync.WaitGroup
+	for _, job := range jobs {
+		wg.Add(1)
+		go test(testClients, job, &wg)
+	}
+	wg.Wait()
+}
+
+// checkoutPR does the checkout and returns whether or not the PR can be merged.
+func checkoutPR(repoPath, clonePath, branch, head string, prNumber int) (bool, error) {
+	cloneCommand := exec.Command("git", "clone", repoPath, clonePath)
+	checkoutCommand := exec.Command("git", "checkout", branch)
+	fetchCommand := exec.Command("git", "fetch", "origin", "pull/"+strconv.Itoa(prNumber)+"/head:pr")
+	mergeCommand := exec.Command("git", "merge", head, "--no-edit")
+	if err := cmdlog.RunWithLogs(cloneCommand); err != nil {
+		return false, err
+	}
+	checkoutCommand.Dir = clonePath
+	if err := cmdlog.RunWithLogs(checkoutCommand); err != nil {
+		return false, err
+	}
+	fetchCommand.Dir = clonePath
+	if err := cmdlog.RunWithLogs(fetchCommand); err != nil {
+		return false, err
+	}
+	mergeCommand.Dir = clonePath
+	if err := cmdlog.RunWithLogs(mergeCommand); err != nil {
+		return false, nil
```

Why not return the err here? Document what `false, nil` means vs `false, err`.